Test 1.0

aksoli666 opened the pull request:
![controller-test-coverage](https://github.com/user-attachments/assets/ab0a2f35-5bd8-467a-b5db-71b1d89a97ed)
![service-test-coverage](https://github.com/user-attachments/assets/723bee1a-cc6d-4968-8d69-08fde19e838a)
added tests for Book repository and tests for service, controller layers for Book and Category

aksoli666 committed:
Test 1.0
added tests for Book repository and tests for service,
controller layers for Book and Category

Elena-Bruyako commented on src/test/java/shop/controller/CategoryControllerTest.java:
```diff
@@ -0,0 +1,258 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CategoryControllerTest {
+    private static final String ADD_CATEGORIES_SQL =
```
it's better to remove these constants and use `@Sql` under the test class with all scripts that you need

Elena-Bruyako commented on src/test/java/shop/service/BookServiceTest.java:
```diff
@@ -0,0 +1,301 @@
+        category.setName(CATEGORY_NAME_CORRECT);
+
+        Book book = new Book();
+        book.setId(ID_1L_CORRECT);
```
check if you have duplication and use separate methods for creating test data 
Also, you can create TestUtil class with methods that return test data and TestConstants for constants

Elena-Bruyako commented on src/test/resources/changelog/db.changelog-master.yaml:
```diff
@@ -0,0 +1,25 @@
+databaseChangeLog:
```
redundant

Elena-Bruyako commented on src/test/resources/database/shop/category/delete-categories.sql:
```diff
@@ -0,0 +1 @@
+DELETE FROM categories WHERE id IN (6, 7);
```
why do you need so many delete scripts?

Elena-Bruyako commented on src/test/resources/database/shop/category/add-category-for-delete-by-id.sql:
```diff
@@ -0,0 +1 @@
+INSERT INTO categories (id, name, description, is_deleted) VALUES (10, "Y Category", "Y Description for Category Y", 0)
```
why do you need so many insert scripts?

aksoli666 committed:
Test 2.0
removed some sql scripts which don`t use
used @sql(before and after test class)
create separated methods in service book, category class ;

aksoli666 committed:
Test 2.0
tried to add sql scripts to pass the github tests
